code/game/objects/items/devices/scanners.dm

```diff
@@ -407,19 +407,19 @@ SLIME SCANNER
 	// Blood Level
 	if(M.has_dna())
 		var/mob/living/carbon/C = M
-		var/blood_id = C.get_blood_id()
-		if(blood_id)
+		var/blood_typepath = C.get_blood_id()
+		if(blood_typepath)
 			if(ishuman(C))
 				if(H.bleed_rate)
 					msg += "<span class='danger'>Subject is bleeding!</span>\n"
 			var/blood_percent =  round((C.scan_blood_volume() / (BLOOD_VOLUME_NORMAL * C.blood_ratio))*100)
 			var/blood_type = C.dna.blood_type
-			if(blood_id != ("blood" || "jellyblood"))//special blood substance
-				var/datum/reagent/R = GLOB.chemical_reagents_list[blood_id]
+			if(blood_typepath != /datum/reagent/blood && blood_typepath != /datum/reagent/blood/jellyblood)
```

Comment:
this probably needs to keep being an Or statement, because not every sample will have both types.

Reply:
The previous statement checked if the blood ID was neither blood nor jellyblood, the statement I replaced it with does the same, I don't even think the previous statement worked as ("blood" || "jellyblood") evaluaes to "blood" because of how the or operator actually works.